### DISCUSSION TITLE
Bug with @swagger_auto_schema in drf-yasg

### DISCUSSION THREAD

**Discussion by Rfank2021** [oxan/djangorestframework-dataclasses#70] [General]
If there are multiple nested different data classes in one data class, the swagger doc will only contain one nested data class type.

We can solve the problem by specifying the nested data class serializer manually. The annotation is here, https://drf-yasg.readthedocs.io/en/stable/custom_spec.html#the-swagger-auto-schema-decorator

**Comment by oxan** [comment]
Do you have any idea if and/or how this could be fixed in djangorestframework-dataclasses? It smells a lot like either a bug in or a bad interaction with drf-yasg (which I've never used).

**Reply by Rfank2021** [reply]
I didn't dig into it.

**Comment by oxan** [comment]
I've converted this to a discussion, as it's related to a 3rd-party package and I don't think it's directly actionable for drf-dataclasses.
